PWGHF/Core/HfMlResponseLcToK0sP.h

```diff
@@ -0,0 +1,300 @@
+/// \author Alexandre Bigot <alexandre.bigot@cern.ch>, IPHC Strasbourg
+/// \author Elisa Meninno <elisa.meninno@cern.ch>, SMI Vienna
+
+#ifndef PWGHF_CORE_HFMLRESPONSELCTOK0SP_H_
+#define PWGHF_CORE_HFMLRESPONSELCTOK0SP_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/Core/HfMlResponse.h"
+
+// Fill the map of available input features
+// the key is the feature's name (std::string)
+// the value is the corresponding value in EnumInputFeatures
+#define FILL_MAP_LC(FEATURE)                                         \
+  {                                                                  \
+#FEATURE, static_cast < uint8_t>(InputFeaturesLcToK0sP::FEATURE) \
+  }
+
+// Check if the index of mCachedIndices (index associated to a FEATURE)
+// matches the entry in EnumInputFeatures associated to this FEATURE
+// if so, the inputFeatures vector is filled with the FEATURE's value
+// by calling the corresponding GETTER from OBJECT
+#define CHECK_AND_FILL_VEC_LC_FULL(OBJECT, FEATURE, GETTER)    \
+  case static_cast<uint8_t>(InputFeaturesLcToK0sP::FEATURE): { \
+    inputFeatures.emplace_back(OBJECT.GETTER());               \
+    break;                                                     \
+  }
+
+// Specific case of CHECK_AND_FILL_VEC_LC_FULL(OBJECT, FEATURE, GETTER)
+// where OBJECT is named candidate and FEATURE = GETTER
+#define CHECK_AND_FILL_VEC_LC(GETTER)                         \
+  case static_cast<uint8_t>(InputFeaturesLcToK0sP::GETTER): { \
+    inputFeatures.emplace_back(candidate.GETTER());           \
+    break;                                                    \
+  }
+
+// Variation of CHECK_AND_FILL_VEC_LC_FULL(OBJECT, FEATURE, GETTER)
+// where GETTER is a method of hfHelper
+#define CHECK_AND_FILL_VEC_LC_HFHELPER(OBJECT, FEATURE, GETTER) \
+  case static_cast<uint8_t>(InputFeaturesLcToK0sP::FEATURE): {  \
+    inputFeatures.emplace_back(hfHelper.GETTER(OBJECT));        \
+    break;                                                      \
+  }
+
+namespace o2::analysis
+{
+// possible input features for ML
+enum class InputFeaturesLcToK0sP : uint8_t {
+  posX,
+  posY,
+  posZ,
+  xSecondaryVertex,
+  ySecondaryVertex,
+  zSecondaryVertex,
+  errorDecayLength,
+  errorDecayLengthXY,
+  chi2PCA,
+  rSecondaryVertex,
+  decayLength,
+  decayLengthXY,
+  decayLengthNormalised,
+  decayLengthXYNormalised,
+  impactParameterNormalised0,
+  ptProng0,
+  impactParameterNormalised1,
+  ptProng1,
+  pxProng0,
+  pyProng0,
+  pzProng0,
+  pxProng1,
+  pyProng1,
+  pzProng1,
+  impactParameter0,
+  impactParameter1,
+  errorImpactParameter0,
+  errorImpactParameter1,
+  v0X,
+  v0Y,
+  v0Z,
+  v0Radius,
+  v0cosPA,
+  v0MLambda,
+  v0MAntiLambda,
+  v0MK0Short,
+  v0MGamma,
+  ctV0,
+  dcaV0daughters,
+  pxPos,
+  pyPos,
+  pzPos,
+  ptV0Pos,
+  dcaPosToPV,
+  pxNeg,
+  pyNeg,
+  pzNeg,
+  ptV0Neg,
+  dcaNegToPV,
+  nSigmaTPCPr0,
+  nSigmaTOFPr0,
```

Comment:
I would also add the combined nSigma variable (see e.g. https://github.com/AliceO2Group/O2Physics/blob/524816143c1d9d4b3ce6f85d6c2e7eed7372bc25/PWGHF/Core/HfMlResponseD0ToKPi.h#L98-L99) Also, don't forget to spawn the tables of the combined nSigma tables (see https://github.com/AliceO2Group/O2Physics/pull/3872)